docs/src/architecture/accounts.md

```diff
@@ -8,44 +8,53 @@ An account is an entity which holds assets and defines rules of how these assets
 In the above:
 
 * **Account ID** is a unique identifier (8 bytes) of an account which does not change throughout its lifetime. It also specifies the type of the underlying account. 
-* **Storage** is user-defined data which can be stored in an account. The data is stored as key-value map which maps 256-bit keys to 256-bit values.
-* **Nonce** is the account's nonce increments every time the account changes its state. 
+* **Storage** is user-defined data which can be stored in an account. 256 values of four field elements each can be stored in a key-value map mapping four field element keys to four field element values.
```

Comment:
This is still not entirely correct and is a bit confusing. Maybe we should just say something simpler here and add a small section which explains how account storage works (we could add a diagram in that section).